### PR TITLE
Confine Angular usage to app.js, separate ostree remote logic

### PR DESCRIPTION
**Pull request by martinpitt**
This is another prerequisite for porting to React.

### CODE REVIEW

**Comment by martinpitt** [comment]
@KKoukiou : Friendly ping?

File: src/remotes.js [hunk 1]
```diff
@@ -97,48 +95,29 @@ angular.module('ostree.remotes', [
...
-                    d.reject(ex);
-                })
-                .always(function () {
-                    file.close();
-                });
+            const file = getRemoteSettingsFile(name);
+            const section = getSectionName(name);
 
-            return d.promise;
+            const promise = file.modify(content => changeData(content, section, options));
```
Comment (KKoukiou) [inline src/remotes.js]: Did you miss the fail handler here?

Comment (KKoukiou) [reply src/remotes.js]: Ah actually it's trivial, not doing anything - sorry for noise.

Comment (martinpitt) [reply src/remotes.js]: Right, the original code just passed it on and just re-wrapped the rejection into a new defer.reject(). I simplified the code to simply pass on the original promise instead, and just kept the .always(file.close) cleanup.

Comment (martinpitt) [reply src/remotes.js]: Please do bring up the "noise" -- that's the point of reviews, to double-check :-) Thanks!

**Pull request merged**